### ISSUE TITLE
fraud_detection:  CPM and CC may dip to negative values

### ISSUE THREAD

**Issue by inderjeetsharma** [OpenSIPS/opensips#2079]
Hello Team,

Im using "opensips 3.0.2 (x86_64/linux)" and getting following concurrent calls.
root@debian:~# opensips-cli -x mi show_fraud_stats 15013880001  1
{
    "cpm": 1,
    "total_calls": 184,
    "concurrent_calls": **4294967295**,
    "seq_calls": 2
}

**Comment by inderjeetsharma** [comment]
Hello Liviuchircu

I 'm still getting this issue.I have tried almost all the latest opensips versions.
opensipsctl fifo show_fraud_stats 1XXXX 1 140XXXXX
calls per minute:: 1
total calls:: 146
concurrent calls:: *4294967292*
sequential calls:: 4


version: opensips 2.4.7 (x86_64/linux)
flags: STATS: On, DISABLE_NAGLE, USE_MCAST, SHM_MMAP, PKG_MALLOC, F_MALLOC,
FAST_LOCK-ADAPTIVE_WAIT
ADAPTIVE_WAIT_LOOPS=1024, MAX_RECV_BUFFER_SIZE 262144, MAX_LISTEN 16,
MAX_URI_SIZE 1024, BUF_SIZE 65535
poll method support: poll, epoll, sigio_rt, select.
git revision: d2f6529
main.c compiled on 06:29:06 Apr 27 2020 with gcc 4.8.5


if (!has_totag()) {
     record_route();
     check_fraud("$fU", "$rU", "1");
     if ($rc == 1) {
     xlog("Rule Match $fU $rU rc=$rc\n");
     }
     xlog("rc $rc\n");
     if ($rc < 0) {
     xlog("Problems for user $fU rc=$rc\n");
     }
     }
     topology_hiding();
       }
          route("RESOLVE_DOMAIN");
     t_on_reply("REPLY");
          route("RELAY");

**Comment by liviuchircu** [comment]
hey @inderjeetsharma, I've raised the priority of this issue.  Expect an update from me by Monday evening, on a worst case.

**Comment by liviuchircu** [comment]
hey @inderjeetsharma, this should now be fixed on supported branches: `2.4` and above.  The problem was caused by internally-generated 408 timeouts vs. 200 OK from the UAS side.  Does this sound familiar to your setup?  Were you seeing any 408 timeouts generated by OpenSIPS at all?

**Comment by inderjeetsharma** [comment]
@liviuchircu Thanks for your response. I will check if get 408 generated by opensips and get back to you as quickly as possible.

**Comment by inderjeetsharma** [comment]
@liviuchircu I cant see any 408 generated by UAS. Should i recompiled with the latest branch?

**Comment by liviuchircu** [comment]
@inderjeetsharma the 408 must not be generated by UAS, rather by OpenSIPS, internally, since the UAS is taking too long.  Yes, you should recompile latest branch.

**Comment by inderjeetsharma** [comment]
@liviuchircu I noticed on more thing that it is not only concurrent calls but same issue coming with calls per minute. Value is coming with -1 minus.
xlog("E_FRD_CRITICAL: $var(param);$var(val);$var(thr);$var(user);$var(number);$var(ruleid)\n");
E_FRD_CRITICAL: concurrent calls;-1;30;1844XXXX844;1423XXXX415;39
E_FRD_CRITICAL: calls per minute;-1;40;1833XXXX706;1210XXXX983;39

**Comment by liviuchircu** [comment]
Let's take the issues one at a time.  So you are saying that 'concurrent calls' is still having issues?  Or are these fraud events from your previous runs, before testing the patch?

**Comment by inderjeetsharma** [comment]
These are old logs day before yesterday. I have recompiled the current version with latest branch.Let's see how it goes from now.I will get back to you with updates.

**Comment by inderjeetsharma** [comment]
Issue is still persists,
Aug  4 14:41:30 localhost /usr/local/sbin/opensips[21148]: E_FRD_CRITICAL: concurrent calls;-1;30;188XXXX134;1412XXXX688;39

**Comment by liviuchircu** [comment]
Can you reliably reproduce this issue and teach me how to do it, @inderjeetsharma ?  If not, could you email your `opensips.cfg` file to liviu@opensips.org?  Thanks!

**Comment by liviuchircu** [comment]
Thank you for the help, @inderjeetsharma - I understand what's going on now, with both cpm and concurrent calls.  The problem is related to interval switching, where the stats are reset to `0` [here](https://github.com/OpenSIPS/opensips/blob/master/modules/fraud_detection/fraud_detection.c#L319), upon calling `check_fraud()` on a newly matching time interval.  The problem is that despite the stats are now reset to `0`, nothing prevents the CPM from still being decremented [here](https://github.com/OpenSIPS/opensips/blob/master/modules/fraud_detection/fraud_detection.c#L371), or the CC from being decremented [here](https://github.com/OpenSIPS/opensips/blob/master/modules/fraud_detection/frd_events.c#L168), after the dialog started in the previous interval finally ends.

Working towards a solution!

**Comment by inderjeetsharma** [comment]
Thanks @liviuchircu

**Comment by inderjeetsharma** [comment]
Hello @liviuchircu Any update.
Thank you.

**Comment by liviuchircu** [comment]
Hey @inderjeetsharma, I was away on holiday last week, but I'm back with renewed powers now.  Could you please apply the following patch for fraud_detection and confirm it fixes the problem for you as well?  It should work on most versions (ideally: 3.0+ version).  Thank you!

```
git apply <(base64 -d <<EOF | gzip -dc
H4sIAH5DOl8AA7VXbW/aSBD+TH7FXKX2ANsEgzFEKFHVJHeqlEtPyVXVvVSrxbvGVo1N10vSXNv/
fjNrGwyBC9wp+RDbM7Ozu88884KIwxAcZxpr4MezTCwSmR+Hii8EE1LLQMdZuvndCWCyv+1RnAr5
BUJvMBiKIfc6HX8kJ64MJj643a7veUeO4xyy+5FlWQed4PVrcHr+yB6CVTxQkGuu4wDiVEMQyeAT
M8uauVaLQEMez9ksn0Ib/9loq6C9yKUqX9PFbEIftLg9j0XrCBqlvyBLc20UKmBSqUzBKTh9mz4D
FaMJT5iOjLRnpPdcpXE6rYSujc7wj1TZp1LqGss0YwpvjN+9MRqFCm+ZTNmcKz7Dc9BjfOQ0is3R
aulhfGRtkeL59R1TMpcaxV10iU7jEJptoVgkuYDTU7h+f3XVgq90puM2XGcguOY2zFU24ZPkAfGI
kwSSjAu8A7SPDdb9Xs/2waLH6DmwbuTSOSOneUdnGgFFUJO8vERdi9EIFkrJVO80yeXnus4qdLib
VHfoORaWZaR1rFzy8d3ghaC8nyMmEnQkzUXzJQqeb7t9hMEb2W7vOXCwrK1AjKuTnZt7zaWCn24u
2O3l+S379fKGfXh7ffHuAx3TMfFOs3sNDqx8JTzXbMY1HlIwHc8knCHnoL3NDZHD+v9u4Nu3FR1X
hMsWOo9FAa7i6VRCFoLuggW9drGQ3b7943IM9xICnv6oASEDnj4Yu0iSG9yQSIv4poHx9ABc4UuW
QZYIgmGdM/PZkiYzOcPjNGtKQpTdY0nL7m3olrmax3/LLGwu0jyeplJAHmUKL7EdL0rRxnaPfxoI
X25b9rGiXePf8D0F8lCysyETxGL/yOwIrqHy0LVdD6k87NHzOVJ6IwLOKezAKP443rRfVy+j9/0J
pF9uw5nSHeEDA15FQ2wzOTKHpwX31piHXol9qcTAa6yPGcQ5vSDblOR5RUFicHFxomZRcULsDiY2
JhX/61FNttdLAeJHJetwjplSt+lso4SuiguF3VwLe4lEyidiWfq8fpcagNd3n6XZEqSm3TlndCax
UKaLYuDxKIWMoZBT+6dO1yl77HjLQmrKuxdWLdu0gHJlrTdAwdKapIDH1EPszZOOktM4Ry116mBC
ssCGi6ufz9+wi8vb327e/V7WkIaIeZLhGCDVLE6xobDzN9hnaUsbQiUlKw1wpZG24AckOmWo2G+G
E0zeYRjzJ8a3lVk5uXG/O/L7cuh2OuGo2/fDyWS47+RWc/bU0FYzJQq5/oDmNXqYGeIuQ7i3YFSR
iCahQGIutvGtYIt+mEt7mSFJhrybYjV/hXuVoTQUd85IZeqy4zzS7cqCouetzNdYcQqP/NT0LSLT
AVuVZ1cyoXqy+/xAFz2UDNF+QYmqMX4wkL57Ep50Ot5kNJCDE94/mAzR/mSIinrimXLiFdWE4ipk
CGXkv5rpW1HnV+PyvSga9LVsymYAxu0oR6118aP8XdPWS8wjj/UyMt4f/DLUT2BQWpXQCx6MeiMR
DDAPRd8X0vWC/aGvfD2NfGVpstAdEfL0MFm41DOMjnpgug2YUoWoWQWhKNxzJcP4S1nE82jGSFW0
+qtf2OXNDTRf4MiWZoQ/AkYk/it90Sq6u5J6gRWdfn6UvdyirGm2a7u3Nmvx2tD2atPWfNHQBuW8
trqMbrWqVnpgCJ9Kn9KqCuGAy0nQ8/xORw5CL+zJE35oCPdJnsqSQjhwKYIDd1fubEgYrY94HrFY
y5lJrqlMmalAGmp/JKl+hpYo1rRG8ijRln/rGff9Ma9MHha5GevKP/7ilE1qBBitfwAit9jBuRAA
AA==
EOF
)
```

**Comment by inderjeetsharma** [comment]
Hello @liviuchircu Thanks for your response. I'm getting following error while applying patch.
[root@testinginder opensips-3.1.0]# git apply <(base64 -d <<EOF | gzip -dc
> H4sIAH5DOl8AA7VXbW/aSBD+TH7FXKX2ANsEgzFEKFHVJHeqlEtPyVXVvVSrxbvGVo1N10vSXNv/
> fjNrGwyBC9wp+RDbM7Ozu88884KIwxAcZxpr4MezTCwSmR+Hii8EE1LLQMdZuvndCWCyv+1RnAr5
> BUJvMBiKIfc6HX8kJ64MJj643a7veUeO4xyy+5FlWQed4PVrcHr+yB6CVTxQkGuu4wDiVEMQyeAT
> M8uauVaLQEMez9ksn0Ib/9loq6C9yKUqX9PFbEIftLg9j0XrCBqlvyBLc20UKmBSqUzBKTh9mz4D
> FaMJT5iOjLRnpPdcpXE6rYSujc7wj1TZp1LqGss0YwpvjN+9MRqFCm+ZTNmcKz7Dc9BjfOQ0is3R
> aulhfGRtkeL59R1TMpcaxV10iU7jEJptoVgkuYDTU7h+f3XVgq90puM2XGcguOY2zFU24ZPkAfGI
> kwSSjAu8A7SPDdb9Xs/2waLH6DmwbuTSOSOneUdnGgFFUJO8vERdi9EIFkrJVO80yeXnus4qdLib
> VHfoORaWZaR1rFzy8d3ghaC8nyMmEnQkzUXzJQqeb7t9hMEb2W7vOXCwrK1AjKuTnZt7zaWCn24u
> 2O3l+S379fKGfXh7ffHuAx3TMfFOs3sNDqx8JTzXbMY1HlIwHc8knCHnoL3NDZHD+v9u4Nu3FR1X
> hMsWOo9FAa7i6VRCFoLuggW9drGQ3b7943IM9xICnv6oASEDnj4Yu0iSG9yQSIv4poHx9ABc4UuW
> QZYIgmGdM/PZkiYzOcPjNGtKQpTdY0nL7m3olrmax3/LLGwu0jyeplJAHmUKL7EdL0rRxnaPfxoI
> X25b9rGiXePf8D0F8lCysyETxGL/yOwIrqHy0LVdD6k87NHzOVJ6IwLOKezAKP443rRfVy+j9/0J
> pF9uw5nSHeEDA15FQ2wzOTKHpwX31piHXol9qcTAa6yPGcQ5vSDblOR5RUFicHFxomZRcULsDiY2
> JhX/61FNttdLAeJHJetwjplSt+lso4SuiguF3VwLe4lEyidiWfq8fpcagNd3n6XZEqSm3TlndCax
> UKaLYuDxKIWMoZBT+6dO1yl77HjLQmrKuxdWLdu0gHJlrTdAwdKapIDH1EPszZOOktM4Ry116mBC
> ssCGi6ufz9+wi8vb327e/V7WkIaIeZLhGCDVLE6xobDzN9hnaUsbQiUlKw1wpZG24AckOmWo2G+G
> E0zeYRjzJ8a3lVk5uXG/O/L7cuh2OuGo2/fDyWS47+RWc/bU0FYzJQq5/oDmNXqYGeIuQ7i3YFSR
> iCahQGIutvGtYIt+mEt7mSFJhrybYjV/hXuVoTQUd85IZeqy4zzS7cqCouetzNdYcQqP/NT0LSLT
> AVuVZ1cyoXqy+/xAFz2UDNF+QYmqMX4wkL57Ep50Ot5kNJCDE94/mAzR/mSIinrimXLiFdWE4ipk
> CGXkv5rpW1HnV+PyvSga9LVsymYAxu0oR6118aP8XdPWS8wjj/UyMt4f/DLUT2BQWpXQCx6MeiMR
> DDAPRd8X0vWC/aGvfD2NfGVpstAdEfL0MFm41DOMjnpgug2YUoWoWQWhKNxzJcP4S1nE82jGSFW0
> +qtf2OXNDTRf4MiWZoQ/AkYk/it90Sq6u5J6gRWdfn6UvdyirGm2a7u3Nmvx2tD2atPWfNHQBuW8
> trqMbrWqVnpgCJ9Kn9KqCuGAy0nQ8/xORw5CL+zJE35oCPdJnsqSQjhwKYIDd1fubEgYrY94HrFY
> y5lJrqlMmalAGmp/JKl+hpYo1rRG8ijRln/rGff9Ma9MHha5GevKP/7ilE1qBBitfwAit9jBuRAA
> AA==
> EOF
> )
**error: patch failed: modules/fraud_detection/fraud_detection.c:430
error: modules/fraud_detection/fraud_detection.c: patch does not apply**

**Comment by liviuchircu** [comment]
Are you on latest `3.1` commit?  Here is some output from my `3.1` tree, which seems to work:

```
[liviu ◄ Z370 opensips-3.1 (3.1)]$ git log -2
commit 671d9919e6a054e2fcd9fed7805f7d1403d79ecc (HEAD -> 3.1, origin/3.1)
Author: OpenSIPS <github@opensips.org>
Date:   Sat Aug 15 23:45:01 2020 +0300

    Rebuild documentation

commit f4c0726c8588d2bc20131c88b7089bd3c1c7fbff
Author: Bogdan-Andrei Iancu <bogdan@opensips.org>
Date:   Fri Aug 14 14:56:11 2020 +0300

    [tm] run the per-branch on reply route also as ONREPLY type
    
    Closes #2201
    
    (cherry picked from commit ad178c5754bc70ce6a406e9517e559c62625dfe4)
[liviu ◄ Z370 opensips-3.1 (3.1)]$ git status
On branch 3.1
Your branch is up to date with 'origin/3.1'.

nothing to commit, working tree clean
[liviu ◄ Z370 opensips-3.1 (3.1)]$ git apply <(base64 -d <<EOF | gzip -dc
> 
> H4sIAH5DOl8AA7VXbW/aSBD+TH7FXKX2ANsEgzFEKFHVJHeqlEtPyVXVvVSrxbvGVo1N10vSXNv/
> fjNrGwyBC9wp+RDbM7Ozu88884KIwxAcZxpr4MezTCwSmR+Hii8EE1LLQMdZuvndCWCyv+1RnAr5
> BUJvMBiKIfc6HX8kJ64MJj643a7veUeO4xyy+5FlWQed4PVrcHr+yB6CVTxQkGuu4wDiVEMQyeAT
> M8uauVaLQEMez9ksn0Ib/9loq6C9yKUqX9PFbEIftLg9j0XrCBqlvyBLc20UKmBSqUzBKTh9mz4D
> FaMJT5iOjLRnpPdcpXE6rYSujc7wj1TZp1LqGss0YwpvjN+9MRqFCm+ZTNmcKz7Dc9BjfOQ0is3R
> aulhfGRtkeL59R1TMpcaxV10iU7jEJptoVgkuYDTU7h+f3XVgq90puM2XGcguOY2zFU24ZPkAfGI
> kwSSjAu8A7SPDdb9Xs/2waLH6DmwbuTSOSOneUdnGgFFUJO8vERdi9EIFkrJVO80yeXnus4qdLib
> VHfoORaWZaR1rFzy8d3ghaC8nyMmEnQkzUXzJQqeb7t9hMEb2W7vOXCwrK1AjKuTnZt7zaWCn24u
> 2O3l+S379fKGfXh7ffHuAx3TMfFOs3sNDqx8JTzXbMY1HlIwHc8knCHnoL3NDZHD+v9u4Nu3FR1X
> hMsWOo9FAa7i6VRCFoLuggW9drGQ3b7943IM9xICnv6oASEDnj4Yu0iSG9yQSIv4poHx9ABc4UuW
> QZYIgmGdM/PZkiYzOcPjNGtKQpTdY0nL7m3olrmax3/LLGwu0jyeplJAHmUKL7EdL0rRxnaPfxoI
> X25b9rGiXePf8D0F8lCysyETxGL/yOwIrqHy0LVdD6k87NHzOVJ6IwLOKezAKP443rRfVy+j9/0J
> pF9uw5nSHeEDA15FQ2wzOTKHpwX31piHXol9qcTAa6yPGcQ5vSDblOR5RUFicHFxomZRcULsDiY2
> JhX/61FNttdLAeJHJetwjplSt+lso4SuiguF3VwLe4lEyidiWfq8fpcagNd3n6XZEqSm3TlndCax
> UKaLYuDxKIWMoZBT+6dO1yl77HjLQmrKuxdWLdu0gHJlrTdAwdKapIDH1EPszZOOktM4Ry116mBC
> ssCGi6ufz9+wi8vb327e/V7WkIaIeZLhGCDVLE6xobDzN9hnaUsbQiUlKw1wpZG24AckOmWo2G+G
> E0zeYRjzJ8a3lVk5uXG/O/L7cuh2OuGo2/fDyWS47+RWc/bU0FYzJQq5/oDmNXqYGeIuQ7i3YFSR
> iCahQGIutvGtYIt+mEt7mSFJhrybYjV/hXuVoTQUd85IZeqy4zzS7cqCouetzNdYcQqP/NT0LSLT
> AVuVZ1cyoXqy+/xAFz2UDNF+QYmqMX4wkL57Ep50Ot5kNJCDE94/mAzR/mSIinrimXLiFdWE4ipk
> CGXkv5rpW1HnV+PyvSga9LVsymYAxu0oR6118aP8XdPWS8wjj/UyMt4f/DLUT2BQWpXQCx6MeiMR
> DDAPRd8X0vWC/aGvfD2NfGVpstAdEfL0MFm41DOMjnpgug2YUoWoWQWhKNxzJcP4S1nE82jGSFW0
> +qtf2OXNDTRf4MiWZoQ/AkYk/it90Sq6u5J6gRWdfn6UvdyirGm2a7u3Nmvx2tD2atPWfNHQBuW8
> trqMbrWqVnpgCJ9Kn9KqCuGAy0nQ8/xORw5CL+zJE35oCPdJnsqSQjhwKYIDd1fubEgYrY94HrFY
> y5lJrqlMmalAGmp/JKl+hpYo1rRG8ijRln/rGff9Ma9MHha5GevKP/7ilE1qBBitfwAit9jBuRAA
> AA==
> EOF
> )
[liviu ◄ Z370 opensips-3.1 (3.1)]$ git status
On branch 3.1
Your branch is up to date with 'origin/3.1'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

	modified:   modules/fraud_detection/fraud_detection.c
	modified:   modules/fraud_detection/frd_events.c
	modified:   modules/fraud_detection/frd_events.h
	modified:   modules/fraud_detection/frd_stats.c
	modified:   modules/fraud_detection/frd_stats.h

no changes added to commit (use "git add" and/or "git commit -a")
[liviu ◄ Z370 opensips-3.1 (3.1)]$ 
```

**Comment by liviuchircu** [comment]
Nevertheless, [here](https://gist.github.com/liviuchircu/ea76c80e4c7b961eae3c9f4c5fce5cac) is the patch, maybe you can just apply it with `patch -p1 < opensips-3.1-fraud-detection-fix-2079.patch` if you can't get `git apply` to work.

**Comment by inderjeetsharma** [comment]
Yes this patch worked with 3.1.0 but i can't get  git apply worked with any other version like 2.4.7, 2.4.8, 3.0.1. Currently im using 2.4.7 in production.I need patch this.

**Comment by liviuchircu** [comment]
Thank you for the details -- please find a version that should apply on `2.4`, fingers crossed!

```
git apply <(base64 -d <<EOF | gzip -dc
H4sIAGGQOl8AA7VXbW/bNhD+7PyKW4FutvUS25H8AsNB0SQbCmTtkKwo9lIQtEjZRGXJo+ikWdv/
vjtKtmXHXuwN8QeLujseyeeeu6OEimPwvIkywE9nmVgkMj+NNV8IJqSRkVFZuv3uRzA+3PZEpUJ+
hoHoRGJ81o99vyfOgrNWS0hot1rdIDjxPO+Y1U8cxzlqB69egdfp9dweOMUDBbnhRkWgUgPRVEaf
mJ1Wz41eRAZyNWezfAJN/HPRgGtoskUu9eolXczGldf5CdRwLhQ2hXKIskWaq0kqhV1orgTJYo3b
SyZszjWfQdM+hidejUx0BCP8Y9knZqbowdkhdUGZO6ZlLg2KW+gSnaoY6k2h2VRyAaMRvH1/fd2A
L6ipnTbhbQaCG+7CXGdjPk4e8PwqSSDJuFApnvPUgnQWdN0uOPToPxNItVx65+Q3901meMIiniR5
eY6qNsrSaKG1TM1ek1z+VdU5hQ53K/UdelbCcay0ClebfHyzkCEu7+cIiwQzlfas+QqIXsttnyES
vY7b7jwTFI6zE4vhcnMX9mhzqeHHm0t2e3Vxy365umEf3ry9fPeBdurZqKfZvQEP1r4Snhs24wb3
KZhRMwnnI+hAc5cboojz/93A169rUq5ply1MrkSBr+bpREIWg2mBA51mMZHdvvn9agj3EiKe/mAo
f4CnD9ZuKskNLkjURYjTyHp6AK5xkGWQJYJg2KTNfLZiykzOcDv1ipIQZfdYkbJ7F1ouGeFk9bfM
4voqVfNppvEQu/GiRK3t9viHhfDlrmkfl8yr/Ru+IyAPJUFrMkEsDo/MnuBaNg9Ctx0gmwddej5T
Ym8FwRvBHpjUx+G2/aZ6FcBvT4D9chfUlPSIIFj8lkzERpEjeXha0G+DfOiVCJhKjL3BQpmBymmA
hNOS50sWEomLcxM7i7oTZ9rGrMjG/7pVm/DVaoD4UeE6nma24G072yqk6/pCkbfHwqYikfWJWBXA
IAyoEwRh+FztklC1rc87p22JhWb3XKcYe9xNIWMo5NTDqev5pMVeNdwxMdLK7J9IWoVi2wvKmZUm
AQVRK5ICIVsVsU+PfS0nKkctde1oTLLIhcvrny5es8ur219v3v1WVpKaUDzJJgxtZyrFzsIuXmPP
pSVdiLWUrDTAmVbagO+Q65Sn4rCLmGDyDiOZP3EHW5uV16+gJWI+6Pci3w8Hg3a/HwYHX78qzp66
eVVMiUXtLtHHoYe9T9xlCPcOjJY8oltRJDEdmzhyLdvMw1y6qyRJMqTeBGv697hWGUrLcu+cVLY6
e94j3b5EKDrf2nyDFSN45KeibxCZjliq3LuWCZWU/fsHOuixZJgeFpRpSYY4DGW3PYgHvh+M+6EM
B/zsaDJMDyfDtCgpRUUJioJCcRUyhjLyXyr352E53nOL1rgc5aizKX6Uvxvaaol55LFaRoaHg1+G
+gkMSqsSesGjfqcvotD3Y3HWFbIdRIdDv/T1NPJLS5uF7T4hTw+bhSs9w+joB2aagClViOrrjxga
zbWM1edi3MynM1vUG7bYXf/Mrm5uoP4CL25pRvgjYETiP9MXjaLBa2kWWNHpU6Rs5w5lTb1ZWb2x
XYs3rm7fb9vaN7q6QXlrWx/GNBrLbnpkCJ9Kn9JqGcKQy3HUCbq+L8M4iDtywI8N4SHJs7SkEIZt
iiD+78mdLQmj+VOeT5kycmaTayJTZiuQgcqPJMtP0hLFitZKHiXa6reZcd8e88rmYZGbyiz949en
rFMjwGj9A0uj4V1+EAAA
EOF
)
```

**Comment by inderjeetsharma** [comment]
Yes it worked when i downloaded opensips from "git clone https://github.com/OpenSIPS/opensips.git -b 2.4 opensips-2.4".

**Comment by inderjeetsharma** [comment]
Patch has been applied to the production server and it will be showing the result in next few hours.

**Comment by inderjeetsharma** [comment]
Everything looks good so far. Thanks @liviuchircu i really appreciate your help.

**Comment by liviuchircu** [comment]
Thank YOU for all your help, @inderjeetsharma.  I will backport this fix down to 2.4, inclusively.  Cheers!
